File: mic_agv/scripts/sub.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"
#include <std_msgs/Int32MultiArray.h>

void chatterCallback(const std_msgs::Int32MultiArray::ConstPtr& msg)
{
  ROS_INFO("I heard: [%s]", msg->data[0].c_str());

  ROS_INFO("I heard: [%s]", msg->data.c_str());
}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "sub");
  ros::NodeHandle n;
  ros::Subscriber sub = n.subscribe("enc_data", 1000, chatterCallback);
  ros::spin();
  return 0;
}

File: mic_agv/scripts/sub.py
#!/usr/bin/env python3
import rospy
from std_msgs.msg import Int32,Float64,Float32MultiArray

import time

enc_list = []

def callback2(data):
    enc_list=data.data
    rospy.loginfo(enc_list)


def sub_array():
    rospy.init_node('node_1',anonymous=True)
    rospy.Subscriber("/enc_data",Float32MultiArray,callback2)
    rospy.spin()


if __name__ == '__main__':
    try:
        sub_array()
    except rospy.ROSInterruptException:
        pass